### PR TITLE
adding tag actions to directory

### PR DESCRIPTION
**Pull request by joshuaroot**
Per #1671 , AWS now supports tagging of Directories.

- adding tag/remove-tag actions to directory

### CODE REVIEW

File: c7n/resources/directory.py [hunk 1]
```diff
@@ -30,6 +34,31 @@ class resource_type(object):
...
 
+    def augment(self, directories):
+
+        def _add_tags(d):
+            client = local_session(self.session_factory).client('ds')
+            try:
+                tags = client.list_tags_for_resource(
+                    ResourceId=d['DirectoryId']).get('Tags', [])
+            except ClientError as e:
+                if e.response['Error']['Code'] == 'AccessDenied':
```
Comment (kapilt) [inline c7n/resources/directory.py]: this error handling doesn't make any sense, if there's no resource policy on a type of resource, then a denied is a persistent error wrt to the custodian execution role permissions, that needs to bubble up. the error handling through the rest of the pr also seems suspect.

Comment (joshuaroot) [reply c7n/resources/directory.py]: ah, yes. you are correct.

**Comment by kapilt** [comment]
@joshuaroot is this still WIP?

**Comment by joshuaroot** [comment]
no, it's ready. got sidetracked and forgot to circle back.

**Pull request merged**